train.js

```diff
@@ -0,0 +1,166 @@
+/*************
+
+MAIN LOGIC FOR POPULATING FIRE BASE AND THE WEBPAGE WITH LATEST TRAIN INFORMATION
+
+**************/
+
+
+// Initialize Firebase
+var config = {
+  apiKey: "AIzaSyC7QVIJx72Uv8Fo-2U_Ece4dzYNw9X1lnI",
+  authDomain: "train-activity-4d0c2.firebaseapp.com",
+  databaseURL: "https://train-activity-4d0c2.firebaseio.com",
+  projectId: "train-activity-4d0c2",
+  storageBucket: "train-activity-4d0c2.appspot.com",
+  messagingSenderId: "422417814962"
+};
+firebase.initializeApp(config);
+
+//Get ref to the database
+var database = firebase.database();
+
+// Variables to udpate Train Table by the minute
+var tableUpdate = 60000; // 1 minute
+var tableUpdateTimer;
+var trainFttArray = [];
+
+// Update the table Next Arrival and Minutes away by the minute 
+function tableUpdateFunction() {
+
+  var numRows = $("table").children()
+      .eq(1) // tbody
+      .children("tr").length;  //  # of rows
+      
+  // For each Row    
+  for (var i=0; i< numRows; i++) {
+    // Get the array of Tds for each row
+    var trainRowTds = $("table").children().eq(1).children("tr").eq(i).children("td");
+    var name = trainRowTds.eq(0).text();
+
+    // Get the corresponding first Train Time from the saved array
+    // There is a better way to do it to improve performance but this works for now
+    for (var i=0; i<trainFttArray.length; i++) {
```

Comment:
This should be changed, technically this code shouldn't work, when you have nested loops and you are tracking the index you need to create different variables so that there aren't issue when referencing different parts of the iterable object.